patroni/api.py

```diff
@@ -140,6 +152,22 @@ def do_GET(self, write_status_code_only=False):
             elif path in ('/async', '/asynchronous') and not is_synchronous:
                 status_code = replica_status_code
 
+        # check for user defined tags in query params
+        if status_code == 200:
```

Comment:
It makes sense to check tags only for non-leader health checks

Reply:
I was thinking if that's what the user wants then let them have it. But if this way is better then I'll change it

Reply:
Tags have nothing to do with application-level sharding.
If you run many shards (clusters), they must have a different cluster name (scope).

IMO, the leader should always be reported as leader, no matter what tags are requested in a health check.